### PR TITLE
chore: add dependabot config

### PR DESCRIPTION
**Pull request by Elfo404**
Adds dependabot config for gh actions, go dependencies and docker.

The config is generate via `make gendependabot` and a check is added to CI to make sure each package is covered.

Fixes https://github.com/grafana/deployment_tools/issues/156586

Note: OpenTelemetry-related dependencies updates are ignored as they need to be performed manually due to the changes needed in `config/manifest.yaml` which is not supported by dependabot

### CODE REVIEW

**Review by iainlane — DISMISSED** [review]
Looks good as it is, but you could simplify this if you use [`directories`](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#directories) - your choice 🙂

**Comment by Elfo404** [comment]
@iainlane  good point! simplified a bunch, mind a sanity check?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    directory: "/"
+    schedule:
+      interval: "weekly"
+  - package-ecosystem: "docker"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+  - package-ecosystem: "gomod"
+    directories:
+      - "/":
```
Comment (iainlane) [inline .github/dependabot.yml:14]: is this `:` ok?

Comment (Elfo404) [reply .github/dependabot.yml]: nope :)

**Review by iainlane — APPROVED** [review]

**Pull request merged**